user-console/src/main/java/org/pentaho/mantle/client/admin/EmailAdminPanel.java

```diff
@@ -30,6 +30,7 @@
 import com.google.gwt.user.client.ui.TextBox;
 import com.google.gwt.user.client.ui.VerticalPanel;
 import com.google.gwt.user.client.ui.Widget;
+import org.pentaho.gwt.widgets.client.utils.string.StringUtils;
```

Comment:
I wouldn't use this package "org.pentaho.gwt.widgets.client.utils.string.StringUtils" it's our code and it deprecated.
Use apache version "org.apache.commons.lang3", I just checked it's available in maven project.

Reply:
The class which is deprecated is `org.pentaho.gwt.widgets.client.utils.StringUtils`. This one is its replacement and is not.

Still, I'm not against using the Apache library. 

I do still have some doubts on what we can use on the GWT side, because not everything can be translated to JS, or not everything is configured to be visible to the GWT compiler. I suspect that's the reason why these Apache-cloned utilities were created in the first place. Maybe this is unfounded, or not applicable anymore. Trying it out will easily clear any doubts.

Reply:
> The class which is deprecated is `org.pentaho.gwt.widgets.client.utils.StringUtils`. This one is its replacement and is not.
> 
> Still, I'm not against using the Apache library.
> 
> I do still have some doubts on what we can use on the GWT side, because not everything can be translated to JS, or not everything is configured to be visible to the GWT compiler. I suspect that's the reason why these Apache-cloned utilities were created in the first place. Maybe this is unfounded, or not applicable anymore. Trying it out will easily clear any doubts.

Did you mean  https://github.com/pentaho/pentaho-platform/blob/master/core/src/main/java/org/pentaho/platform/util/**StringUtil.java** _is not deprecated_

but https://github.com/pentaho/pentaho-commons-gwt-modules/blob/master/widgets/src/main/java/org/pentaho/gwt/widgets/client/utils/**StringUtils.java**#L26
**_is deprecated_**

Reply:
@dcleao you are correct compiling with apache's StringUtils causes a build error https://github.com/pentaho/pentaho-platform/pull/5185/commits/00da0fe08d4aa9bd4587c2fc43e967e045441341

and when I tried out locally using https://github.com/pentaho/pentaho-platform/blob/master/core/src/main/java/org/pentaho/platform/util/StringUtil.java that caused a build error as well.

A previous architect mentioned in the past to use StringUitl.java : https://github.com/pentaho/pentaho-commons-gwt-modules/blob/04329382317b1c5207ffbacf0df237614df00183/pentaho-gwt-widgets/src/org/pentaho/gwt/widgets/client/utils/StringUtils.java#L21-L27


My mistake @soagarwal1 using https://github.com/pentaho/pentaho-commons-gwt-modules/blob/master/widgets/src/main/java/org/pentaho/gwt/widgets/client/utils/StringUtils.java#L26 was the correct library.

Reply:
@soagarwal1  my mistake. revert back to using  **utils.string.StringUtils**. Using apache's StringUtils will revert in a compile error : https://github.com/pentaho/pentaho-platform/pull/5185/commits/00da0fe08d4aa9bd4587c2fc43e967e045441341

I didn't realize there were two pentaho versions of StringUtil, I Just picked the first one that popped up in my IDE.

Reply:
I created [BISERVER-14898](https://jira.pentaho.com/browse/BISERVER-14898)